tensorflow/contrib/ignite/python/kernel_tests/ignite_test.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License.  You may obtain a copy of
+# the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+# ==============================================================================
+"""Tests for IgniteDataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.data.ops import iterator_ops
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import errors
+from tensorflow.python.ops import array_ops
+from tensorflow.python.platform import test
+from functools import reduce
+import tensorflow as tf
+from tensorflow.contrib.ignite.python.ops import ignite_dataset_ops
+
+class IgniteDatasetTest(test.TestCase):
+    def setUp(self):
+        # The Ignite server has to be setup before the test
+        # and tear down after the test manually.
+        # The docker engine has to be installed.
+        #
+        # To setup the Ignite server:
+        # $ bash ignite_test.sh start ignite_test
+        #
+        # To team down the Ignite server:
+        # $ bash ignite_test.sh stop ignite_test
+        # LD_LIBRARY_PATH should be set in such a way that it contains libjvm.so, for example
+        # export LD_LIBRARY_PATH=/usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/
+        # IGNITE_HOME should point to Apache Ignite installation directory, for example
+        # export IGNITE_HOME=~/apache-ignite-fabric-2.4.0-bin
+        # TF_IGNITE_CLIENT_CONFIG should point to client ignite node config, for example
+        # export TF_IGNITE_CLIENT_CONFIG=../../sample_configs/client.xml
+        pass
+
+    def testIgniteDataset(self):
+        expected_caches = self.collect_values(["node1_entries/entries.txt", "node2_entries/entries.txt"])
+        for cache in expected_caches:
+            self.tst_cache(expected_caches[cache], cache)
+
+    def tst_cache(self, expected_values, cache_name):
+        values = {}
+
+        size = reduce(lambda x, value: x + value, expected_values.values(), 0)
+
+        ds = ignite_dataset_ops.IgniteDataset(cache_name)
+
+        with tf.Session() as sess:
+            it = ds.make_one_shot_iterator()
+            for i in range(0, size):
+                val = sess.run(it.get_next());
```

Comment:
Nit: do not call `Iterator.get_next()` in a loop. Instead call it outside the loop to get a symbolic tensor, and pass that same tensor to `sess.run()` in each iteration.

Reply:
Done.